src/test/java/bio/terra/service/profile/ProfileServiceUnitTest.java

```diff
@@ -0,0 +1,155 @@
+        );
+    }
+
+    @Test
+    public void testCreateProfile() {
+        var billingProfileRequestModel = new BillingProfileRequestModel();
+        billingProfileRequestModel.setProfileName("name");
+
+        var user = new AuthenticatedUserRequest();
+
+        var jobBuilder = mock(JobBuilder.class);
+
+        when(jobBuilder.submit()).thenReturn("id");
+
+        when(jobService.newJob(eq("Create billing profile 'name'"),
+                eq(ProfileCreateFlight.class), any(), any())).thenReturn(jobBuilder);
+
+        String result = profileService.createProfile(billingProfileRequestModel, user);
+        verify(jobBuilder, times(1)).submit();
+        assertEquals(result, "id");
+    }
+
+    @Test
+    public void testUpdateProfile() {
+        var billingProfileUpdateModel = new BillingProfileUpdateModel();
+        billingProfileUpdateModel.setId("name");
+
+        var user = new AuthenticatedUserRequest();
+
+        var jobBuilder = mock(JobBuilder.class);
+
+        when(jobBuilder.submit()).thenReturn("id");
+
+        when(jobService.newJob(eq("Update billing for profile id 'name'"),
+                eq(ProfileUpdateFlight.class), any(), any())).thenReturn(jobBuilder);
+
+        String result = profileService.updateProfile(billingProfileUpdateModel, user);
+        verify(iamService, times(1))
+                .verifyAuthorization(
+                        eq(user),
+                        eq(IamResourceType.SPEND_PROFILE),
+                        any(),
+                        eq(IamAction.UPDATE_BILLING_ACCOUNT)
+                );
+        verify(jobBuilder, times(1)).submit();
+        assertEquals(result, "id");
+    }
+
+    @Test
+    public void testDeleteProfile() {
+        var jobBuilder = mock(JobBuilder.class);
+
+        when(jobBuilder.submit()).thenReturn("id");
+        when(jobBuilder.addParameter(eq(ProfileMapKeys.PROFILE_ID), eq("name"))).thenReturn(jobBuilder);
+
+        when(jobService.newJob(eq("Delete billing profile id 'name'"),
+                eq(ProfileDeleteFlight.class), any(), any())).thenReturn(jobBuilder);
+
+        var user = new AuthenticatedUserRequest();
+        String result = profileService.deleteProfile("name", user);
+        verify(iamService, times(1))
+                .verifyAuthorization(
+                        eq(user),
+                        eq(IamResourceType.SPEND_PROFILE),
+                        any(),
+                        eq(IamAction.DELETE)
+                );
+        verify(jobBuilder, times(1)).submit();
+        assertEquals(result, "id");
+    }
+
+    @Test
+    public void testVerifyAccountHasAccess() {
+        var user = new AuthenticatedUserRequest();
+        String id = "id";
+
+        when(googleBillingService.canAccess(any(), eq("id"))).thenReturn(true);
+
+        try {
+            profileService.verifyAccount(id, user);
+        } catch (Exception e) {
+            fail("No exception should've been thrown");
+        }
```

Comment:
In this case I would add `throws Exception` to the method and skip the try/catch. In general I don't think it's helpful to `catch` in a test, unless it's a case where an exception is not a failure.

Reply:
I don't think I need the throws Exception right? The test will fail if an exception occurs.

Reply:
Well, I was suggesting to replace the try/catch with `profileService.verifyAccount(id, user);` and add `throws Exception` to the method signature, if needed. (I didn't check to see if `verifyAccount()` throws a checked exception, if it doesn't then yes, there's no need to add a `throws` clause to the method signature.)